pkg/manager/internal.go

```diff
@@ -606,6 +623,15 @@ func (cm *controllerManager) engageStopProcedure(stopComplete <-chan struct{}) e
 }
 
 func (cm *controllerManager) startLeaderElectionRunnables() error {
+	for _, hook := range cm.prestartHooks {
```

Comment:
Since this runs synchronously, this is quite the footgun and should have a configurable timeout. We should also log when we start this.

Reply:
Makes sense, I'll update the PR.